martian/src/lib.rs

```diff
@@ -273,7 +277,7 @@ fn martian_entry_point<S: std::hash::BuildHasher>(
 
         // write message and stack trace, exit code = 1;
         Err(e) => {
-            report_error(&mut md, &e, is_error_assert(&e));
+            report_error(&mut md.borrow_mut(), &e, is_error_assert(&e));
```

Comment:
The compiler needs a hint here because the return type of `borrow_mut()` is not just `&mut T` but a wrapper type, so it needs a reference for it to automatically deref correctly.